### DISCUSSION TITLE
Customizer/Engineering Community of Practice

### DISCUSSION THREAD

**Discussion by mcruz-umich** [collectiveaccess/providence#1456] [General]
Hello all,

I work at the University of Michigan and am onboarding to help support our collection managers that use "Collective Access".

I would like to start up a group of people that are interested in working together to help solve challenges each of us might encounter with CA.

Here are the topics:

1. Process Definition - How to do things that we tried but are not quite working right
2. Configurations: How to configure the system to turn on/off features, rearrange things, etc
3. Customizations: Creating and link tables to forms, layouts, etc [Implementing all the magical abilities listed in the documentation]
4. Extending the code base - Community driven voting or contributions to extend code base

We could use this group to email each other questions, to discuss challenges, or to help solve issues.

Please let me know if you are interested by replying to this discussion.

Thanks!
Matthew

**Comment by poplarmedia** [comment]
Hello Matt,
I am very interested. I am also 2 hours west of you in Grand Rapids.
Matt P

**Reply by poplarmedia** [reply]
I would like us to include this point raised by Seth last week around supporting release management: 
https://github.com/collectiveaccess/providence/issues/1413#issuecomment-1601008129

**Reply by poplarmedia** [reply]
Also improving documentation.

**Reply by mcruz-umich** [reply]
Excellent.... I will give this discussion topic a couple of weeks to gather members and then reach out and we can arrange a video chat.

**Reply by greermartin** [reply]
I work at Loyola University Chicago Libraries and we recently started using Collective Access and would be very interested in joining!

**Reply by kehh** [reply]
I work for a consulting company (@gaiaresources) in Western Australia and am interested in this.

**Reply by mcruz-umich** [reply]
Hi @poplarmedia, @greermartin, @kehh 

Please email me at mcruz@umich.edu so that I can know your email address and set up a video meeting for us to introduce ourselves and get a discussion started.

Thanks!

**Reply by mbutel** [reply]
Hey, I just saw this, I work with CA in Argentina and Latinamerica, we have our group bibliohack. Would love to join a space like this. I thought there was a slack or sth like that?

**Comment by mcruz-umich** [comment]
Hello! I got loaded down with another non-collective access project that I
am trying to finish up. Also my leadership has gone back and forth on how
we are going to approach supporting CA so there has not been a clear
direction. I believe they will still encourage this user-group to get
started. One option that we are considering is using 3rd party customizers.
I did some research on it that you might find useful:

*Third-Party Customizers*

   1. https://singhconsultinggroup.com/collective-access/
   2.
   https://www2.archivists.org/consultants/janine-biunno-archival-consulting-jlb-archival
   3.
   https://bits.ashleyblewer.com/blog/2018/11/11/installing-collectiveaccess-on-a-cloud-server/


Keep in touch. If you do not hear from me by the end of December please
reach out again.

Sincerely,
Matthew

On Tue, Nov 28, 2023 at 4:32 PM mbutel ***@***.***> wrote:

> Hey, I just saw this, I work with CA in Argentina and Latinamerica, we
> have our group bibliohack. Would love to join a space like this. I thought
> there was a slack or sth like that?
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/collectiveaccess/providence/discussions/1456#discussioncomment-7697378>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AIWEL5FR4CNZPHD6LT6J5WTYGZJ7TAVCNFSM6AAAAAAZYW5Q7GVHI2DSMVQWIX3LMV43SRDJONRXK43TNFXW4Q3PNVWWK3TUHM3TMOJXGM3TQ>
> .
> You are receiving this because you authored the thread.Message ID:
> ***@***.***
> .com>
>


-- 
Matthew Cruz, Application Architect
Web and Application Development Services
LSA Technology Services
The University of Michigan

**Comment by mbutel** [comment]
Thanks!
Matías

**Comment by goetzk** [comment]
> Hey, I just saw this, I work with CA in Argentina and Latinamerica, we have our group bibliohack. Would love to join a space like this. I thought there was a slack or sth like that?


> I would like to start up a group of people that are interested in working together to help solve challenges each of us might encounter with CA.

There are a number of places to chat about CA but the proposal here sounds like an email discussion list.
CA communications channels I'm aware of:
* Gitter chat
* Matrix.org bridge of Gitter
* CA Developers slack (invite only? I'm not sure but have seen it mentioned)
* Forum https://collectiveaccess.org/support/

Perhaps that last option is worth a look for your purposes?

ps. I just found https://collectiveaccess.org/contact/ does list Slack, so i think the list above is complete.

**Reply by mcruz-umich** [reply]
Thanks @goetzk !

Prior to sending out this information, I had reached out to the CA administration about if there was an existing community and they said there was not so I am going to try to start one up..... The CA-team is pretty loaded down with work so a community-driven group would probably be our best bet for more throughput....

**Reply by collectiveaccess** [reply]
I'm not sure what "community" means in this context. As @goetzk wrote, there are plenty of places to ask questions if you want to ask them. I don't know that another is needed. The Slack channel is open to all, but you do need to ask for an invitation link.

**Comment by mcruz-umich** [comment]
Thank you all. In this context, "community" refers to a group of active users and customizers that could meet with a degree of frequency to solve issues together. It would likely follow some sort of calendar schedule, showcase past, current, and future issues and have a portion of the meetings to focus on on-demand problem-solving. The goal would be to take the best-of information from the resources found online [see the list above from @goetzk] and combine it with our direct experiences, to help each other advance more quickly through our backlogs of customizations, etc. It aims to have a group of us working and talking together real-time rather than sole dependence on forum postings or support-ticket requests. This is just in the early stages of formation so any input from others on what they would like to see or do with this group is appreciated!

I will try to get this rolling early in 2024. In the meantime, what is the process to get an invitation to the slack channel, please?

Thanks!

**Reply by collectiveaccess** [reply]
There's already a group of this sort for Canadian users. I don't know how active they are or how often they meet, but they do meet.

There is a link on this [page](https://collectiveaccess.org/contact/) to request access to Slack.

**Comment by mcruz-umich** [comment]
Thanks!

I do not have an email under any of the allowed domains for joining that slack workspace.... What should I do?
<img width="481" alt="image" src="https://github.com/collectiveaccess/providence/assets/36455924/59fbbd38-298c-4677-907c-9d17ea0508df">

**Comment by mbutel** [comment]
Same problem here!

**Comment by miqrogroove** [comment]
I've been tinkering with CA and PHP 7.4 mainly to get the default theme working and displaying some of the missing fields.  I've written a few PRs for the awful implementation of jcarousel in the object details view, but my strengths are in PHP and MySQL.  I've also had a few PRs accepted to resolve defects in the dacs_heritage profile.

I might be more of a lurker on process definitions and code extension, but I'm generally interested in reading about the DIY efforts.
